src/melee.cpp

```diff
@@ -1366,6 +1416,16 @@ bool player::can_weapon_block() const
             weapon.has_technique( WBLOCK_3 ));
 }
 
+bool player::can_armor_block() const
```

Comment:
You could return a reference to the item here. You will need it later on.
If there is nothing, return `ret_null`. You can then check if the item returned exists by checking if it isn't `is_null()`, like this:
```
item &shield = get_shield();
if( !shield.is_null() ) {
    // Block with shield
}
```